### PR TITLE
feat!: add `cosmwasmAPIGenerator` to keeper

### PR DESCRIPTION
**Pull request by da1suk8**
## Description

closes: #38 

In #36, x/wasm keeper was implemented as a dependency on x/wasmplus.
**This meant that x/wasm could not run alone.**

We added `cosmwasmAPIGenerator` to keeper in order to add dynamiclink functionality as a default function of x/wasm.
This allows the methods of `cosmwasmAPIGenerator` to determine which API to use.

In other words, it is now possible to change the behavior of the APIs within x/wasm and x/wasmplus.

`cosmwasmAPIImpl` and `cosmwasmAPIGeneratorImpl` coexist and the logic is difficult to understand, so I will address this in issue #42

## Motivation and context

Because x/wasm was an implementation that depended on x/wasmplus, 
we wanted to eliminate the dependency and change the contents of the API for x/wasm and x/wasmplus, respectively.

## How has this been tested?

```
make build
make test
```
## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/Finschia/wasmd/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/wasmd/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [x] I have added tests to cover my changes.
- [x] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

**Review by loloicci — DISMISSED** [review]

**Review by kokeshiM0chi — APPROVED** [review]

**Review by dudong2 — DISMISSED** [review]

File: x/wasm/keeper/exported_events.go [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
 
+import (
+	sdk "github.com/line/lbm-sdk/types"
+	wasmvmtypes "github.com/line/wasmvm/types"
+)
+
 var (
 	ContractSDKEventAttributes = contractSDKEventAttributes
 	EventTypeMinLength         = eventTypeMinLength
 )
```
Comment (zemyblue) [inline x/wasm/keeper/exported_events.go:11]: This is last changed, but it is not used. If not using, how about removing it?

File: x/wasm/keeper/events.go [hunk 2]
```diff
@@ -65,3 +65,39 @@ func contractSDKEventAttributes(customAttributes []wasmvmtypes.EventAttribute, c
...
+	return events, nil
+}
+
+func callablePointSDKEventAttributes(customAttributes []wasmvmtypes.EventAttribute, contractAddr sdk.AccAddress, callstack []byte) ([]sdk.Attribute, error) {
+	attrs, err := contractSDKEventAttributes(customAttributes, contractAddr)
+	if err != nil {
+		return nil, err
+	}
+	// attrs[0] is addr
+	attrs = append([]sdk.Attribute{attrs[0], sdk.NewAttribute(types.AttributeKeyCallstack, string(callstack))}, attrs[1:]...)
```
Comment (zemyblue) [inline x/wasm/keeper/events.go:101]: Is there a reason to put callstack first? It's kind of weird.

Comment (da1suk8) [reply x/wasm/keeper/events.go:101]: @loloicci Could you please answer this question?

Comment (loloicci) [reply x/wasm/keeper/events.go:101]: I did not care when I added this function. If it is better the call stack is in the tail of the attributes, I will leave it in another issue.

**Review by zemyblue — APPROVED** [review]

**Review by loloicci — APPROVED** [review]

File: x/wasm/keeper/events.go (hunk 2 above)
Comment (zemyblue) [reply x/wasm/keeper/events.go:101]: Please check it and then if it is necessary, please add another issue, @loloicci .

Comment (da1suk8) [reply x/wasm/keeper/events.go:101]: I left issue https://github.com/Finschia/wasmd/issues/45.

**Review by dudong2 — APPROVED** [review]

**Pull request merged**